src/components/MultiHopTrade/components/TradeConfirm/hooks/useThorStreamingProgress.tsx

```diff
@@ -76,83 +76,79 @@ export const useThorStreamingProgress = ({
   const swap = useAppSelector(state => selectSwapById(state, swapByIdFilter))
 
   const { sellTxHash, metadata } = swap ?? {}
   const { streamingSwapMetadata } = metadata ?? {}
 
-  useEffect(() => {
-    // don't start polling until we have a tx
-    if (!sellTxHash) return
-    if (!swap) return
-    if (!swap.isStreaming) return
-
-    poll({
-      fn: async () => {
-        const updatedStreamingSwapData = await getThorchainStreamingSwap(sellTxHash, swapperName)
-
-        // no payload at all - must be a failed request - return
-        if (!updatedStreamingSwapData) return
-
-        // We don't have a quantity in the streaming swap data, and we never have.
-        if (!streamingSwapDataRef.current?.quantity && !updatedStreamingSwapData.quantity) {
-          // This is a special case where it _is_ a streaming swap, but it's one of the following cases:
-          // - optimized over a single block, and so doesn't actually stream.
-          // - the streaming swap metadata isn't ready yet
-          //
-          // In both cases its simpler to exit and not render the progress bar.
-          return
-        }
-
-        // thornode returns a default empty response once the streaming is complete
-        // set the count to the quantity so UI can display completed status
-        if (streamingSwapDataRef.current?.quantity && !updatedStreamingSwapData.quantity) {
-          const completedStreamingSwapData: ThornodeStreamingSwapResponseSuccess = {
-            ...streamingSwapDataRef.current,
-            count: streamingSwapDataRef.current.quantity,
+  useQuery({
+    queryKey: ['streamingSwapData', confirmedSwapId, swapperName],
+    queryFn:
+      swap &&
+      swap.swapperName === swapperName &&
+      sellTxHash &&
+      swap.isStreaming &&
+      swap.status === SwapStatus.Pending &&
+      confirmedSwapId
+        ? async () => {
+            const updatedStreamingSwapData = await getThorchainStreamingSwap(
+              sellTxHash,
+              swapperName,
+            )
+
+            // no payload at all - must be a failed request - return
+            if (!updatedStreamingSwapData) return
+
+            // We don't have a quantity in the streaming swap data, and we never have.
+            if (!streamingSwapDataRef.current?.quantity && !updatedStreamingSwapData.quantity) {
+              // This is a special case where it _is_ a streaming swap, but it's one of the following cases:
+              // - optimized over a single block, and so doesn't actually stream.
+              // - the streaming swap metadata isn't ready yet
+              //
+              // In both cases its simpler to exit and not render the progress bar.
+              return
+            }
+
+            // thornode returns a default empty response once the streaming is complete
+            // set the count to the quantity so UI can display completed status
```

Comment:
FYI @NeOMakinG this is the updated total mentioned above, so it looks like we are indeed always upserting correct count if I'm reading this correctly?